src/webviews/server.ts

```diff
@@ -29,48 +30,54 @@ export function registerWebviewServer(webview: vscode.Webview, commands: Protoco
         }
     }
 
-    const messageListener = webview.onDidReceiveMessage(async (event: Message) => {
-        const { id, command, data } = event
-        const metadata: Omit<Message, 'id' | 'command' | 'data'> = {}
+    const messageListener = webview.onDidReceiveMessage(
+        // XXX: In earlier versions of Node the first parameter was used as
+        // the `thisArg` for calling the bound function.
+        //
+        // Fixed in https://github.com/nodejs/node/commit/324a6c235a5bfcbcd7cc7491d55461915c10af34
+        AsyncResource.bind(async function (this: any, event: Message) {
```

Comment:
I almost didn't notice this bug. I'm surprised that Node didn't have any tests that execute the bound function with parameters.

Reply:
Nice find! I wonder if this explains https://github.com/aws/aws-toolkit-vscode/pull/3315#issuecomment-1506896655